Support editing title, description and summary of a saved search

linuspahl opened the pull request:
This PR implements the "Edit metadata" action for the saved searches search bar menu, which opens a modal to edit the title, description and summary.

It also:
* renames the "Edit" dashboard search bar action to "Edit metadata"
I think "Edit" is too generic. From a user's perspective: What is going to happen, will a click on the button display the search / widgets in an edit mode?
* adjusts the sidebar view description.
For an ad hoc search, an info will be displayed to save the search as a saved search / dashboard, to add a description / summary. On a dashboard / saved search the view description will include a view type specific placeholder, if a summary or description is not defined.

Regarding the tests for `SavedSearchControls`, I've tried to test if the edit modal opens, when clicking on "Edit Metadata", but I could not easily test this case with `enzyme`. Out of Interest I would like to have another look with someone else.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

kmerz commented on graylog2-web-interface/src/views/components/views/ViewPropertiesModal.jsx:
```diff
@@ -23,7 +24,8 @@ export default class ViewPropertiesModal extends React.Component {
   }
 
-  componentWillReceiveProps(nextProps) {
+  // eslint-disable-next-line camelcase
```
I does not have to be in this PR, but wouldn't it be wiser to make a functional component of this component, instead of adding a UNSAFE_

kmerz commented on graylog2-web-interface/src/views/components/views/ViewPropertiesModal.jsx:
```diff
@@ -23,7 +24,8 @@ export default class ViewPropertiesModal extends React.Component {
+  UNSAFE_componentWillReceiveProps(nextProps) {
     const { title } = this.props;
     const { view } = this.state;
     if (title !== nextProps.title || !isEqual(view, nextProps.view)) {
```
When we turn this into a functional component, we should do something better then comparing the whole view.
1. it is not very efficient
2. we do not really care about the whole view if I am not wrong

kmerz commented on graylog2-web-interface/src/views/components/sidebar/ViewDescription.jsx:
```diff
@@ -1,46 +1,55 @@
       <Section>
-        <SearchResultOverview results={results} />
+        {viewMetadata.description || <i>No {viewTypeLabel} description.</i>}
       </Section>
```
The order of Sections does not make much sense to me:
![Graylog](https://user-images.githubusercontent.com/448763/82921466-b16faf80-9f78-11ea-815b-fc0b8bf96815.png)

I would rather have the description and summary together and the result in the beginning or the end.

linuspahl commented:
@kmerz I like the suggestion with the changed order and moved the search execution info above the summary and description.

I am also a fan of transforming `ViewPropertiesModal` to a functional component and added this task to my refactoring list.
